library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/Section.java

```diff
@@ -39,11 +40,13 @@
      * @param itemResourceId layout resource for its items
      * @param loadingResourceId layout resource for its loading state
      * @param failedResourceId layout resource for its failed state
+     * @param emptyResourceId layout resource for its empty state
      */
-    public Section(int itemResourceId, int loadingResourceId, int failedResourceId) {
```

Comment:
It's a great book!

We can apply the technique in the following way:

1) Create class `SectionParameters`;
2) Create Builder class for `SectionParameters`;
3) Create constructor for `Section` with one parameter: `SectionParameters`;

So in the constructor of your CustomSection class (that extends Section/StatelessSection), you will have to build a SectionParameters class to call super(SectionParameters).

Please let me know if you need more clarification.